radixdlt-engine/src/main/java/com/radixdlt/atom/TxAction.java

```diff
@@ -64,5 +64,216 @@
 
 package com.radixdlt.atom;
 
-/** Marker interface for RadixEngine actions. */
-public interface TxAction {}
+import static com.radixdlt.utils.Strings.asEmptyIfNull;
+import static java.util.Objects.requireNonNull;
+
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.identifiers.REAddr;
+import com.radixdlt.utils.Bytes;
+import com.radixdlt.utils.UInt256;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.function.LongFunction;
+
+/** RadixEngine actions. */
+public sealed interface TxAction {
+  record BurnToken(REAddr resourceAddr, REAddr fromAddr, UInt256 amount) implements TxAction {
+    public BurnToken {
+      ensureNonZeroAmount(amount);
+    }
+  }
+
+  record CreateFixedToken(
+      REAddr resourceAddr,
+      REAddr accountAddr,
+      String symbol,
+      String name,
+      String description,
+      String iconUrl,
+      String tokenUrl,
+      UInt256 supply)
+      implements TxAction {
+    public CreateFixedToken {
+      ensureResourceAddressIsHashedKey(resourceAddr);
+      ensureNonZeroSupply(supply);
+      requireAllNonNull(accountAddr, symbol, name, description, iconUrl, tokenUrl);
```

Comment:
I'd prefer to have separate lines for `requireNonNull`

Reply:
Done